tools/wptrunner/wptrunner/testharnessreport.js

```diff
@@ -69,7 +69,7 @@ window.__wptrunner_process_next_event = function() {
   var props = {output: %(output)d,
                timeout_multiplier: %(timeout_multiplier)s,
                explicit_timeout: %(explicit_timeout)s,
-               message_events: ["completion"]};
+               message_events: ["start", "test_state", "result", "completion"]};
```

Comment:
I'd expect this change wouldn't be needed in this "new approach"?

Reply:
Damn, I thought I had removed the earlier changes in `Try different approach` :(. Well that invalidates my test run...

Reply:
Oh, should be good to review now. Ignore the second patchset touching all the OP files, I will be reverting that.

I think the WPT check results look good from a quick pass.